crates/bevy_ecs/src/query/state.rs

```diff
@@ -276,16 +276,10 @@ where
         }
     }
 
-    /// Returns an [`Iterator`] over all possible combinations of `K` query results without repetition.
-    /// This can only be called for read-only queries.
-    ///
-    ///  For permutations of size K of query returning N results, you will get:
-    /// - if K == N: one permutation of all query results
-    /// - if K < N: all possible K-sized combinations of query results, without repetition
-    /// - if K > N: empty set (no K-sized combinations exist)
+    /// Returns an [`Iterator`] over the query results for the given [`World`] without updating the query's archetypes.
+    /// Archetypes must be manually updated before by using [`Self::update_archetypes`].
     ///
-    /// This can only be called for read-only queries, see [`Self::iter_combinations_mut`] for
-    /// write-queries.
+    /// This can only be called for read-only queries.
```

Comment:
Why do you the changes of #3644 present here?

Reply:
To be honest, this is my first time trying to contribute to an open source project. I'll try to find a way to remove this commit from the pull request.